### PR TITLE
org management

### PR DESCRIPTION
**Pull request by adammeyer**

### CODE REVIEW

File: app/assets/javascripts/angular/components/orgManagement/orgManagement.scss [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+
+    $wrapper-fixed-width: 1112px;
+
+    .wrapper {
+        width: $wrapper-fixed-width;
+        margin: 0 auto;
+        text-align: left;
+    }
+
+    .bg-pivot-gray-5 {
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/orgManagement/orgManagement.scss]: Can you remove this class and add it here https://github.com/CruGlobal/missionhub-web/blob/master/app/assets/stylesheets/variables/colors.scss#L43 ? That'll auto generate that class.

File: app/assets/javascripts/angular/components/orgManagement/orgManagementEdit.component.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+function orgManagementEditController(organizationService) {
+    this.helpIcon = helpIcon;
+
+    this.$onInit = () => {
+        if (this.new) {
+            this.orgEdit = {
+                show_sub_orgs: true,
+            };
+        } else {
+            this.orgEdit = angular.copy(this.org);
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/orgManagement/orgManagementEdit.component.js]: `this.orgEdit = { ...this.org };` will do this too.

File: app/assets/javascripts/angular/components/orgManagement/orgManagementEdit.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    <h3 class="f3 fw6 pivot-dark-blue">
+        {{$ctrl.new ? 'common:organizations.new' : 'common:organizations.edit' | t}}
+    </h3>
+    <fieldset class="ba b--transparent ph0 mh0 w-100">
+        <div class="flex justify-between mb3">
+            <div class="w-48">
+                <label class="db f6 fw6 lh-copy normal mb1 pivot-grey-darker ttu tracked flex items-center"
+                       for="ministryName">
+                    {{'common:organizations.name' | t}}
+                    <a class="help" href="">
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/orgManagement/orgManagementEdit.html]: Is this supposed to go somewhere?

File: app/assets/javascripts/angular/components/orgManagement/orgManagementEdit.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                        <img alt="Help" ng-src="{{$ctrl.helpIcon}}" class="ml1 db" style="width:16px; height:16px;">
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/orgManagement/orgManagementEdit.html]: I feel like alt attributes should be translated but idk what the most efficient way to do that is...

File: app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            organizationOverviewSuborgsService
+                .loadOrgSubOrgs(this.org)
+                .then(subOrgs => {
+                    this.subOrgs = subOrgs;
+                });
+        }
+    };
+
+    this.editSubOrg = ($event, orgId) => {
+        $event.stopPropagation();
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js]: Why is this needed?

File: app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        this.newExpanded[orgId] = !this.newExpanded[orgId];
+    };
+
+    this.editComplete = (orgId, refresh) => {
+        this.editExpanded[orgId] = false;
+        this.newExpanded[orgId] = false;
+
+        if (refresh) {
+            this.orgExpanded[orgId] = false;
+            $timeout(() => {
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js]: This seems hacky...

File: app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+        confirmModalService
+            .create(
+                tFilter('ministries:organizations.delete.confirm', {
+                    org_name: org.name,
+                }),
+            )
+            .then(() => {
+                organizationService.deleteOrg(org).then(() => {
+                    this.subOrgs.splice(this.subOrgs.indexOf(org), 1);
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js]: A filter looking for all items that don't match that id might read better... Idk. This is fine. I guess a splice is probably more performant.

File: app/assets/javascripts/angular/components/orgManagement/orgManagementEdit.html (hunk 3 above)
Comment (adammeyer) [reply app/assets/javascripts/angular/components/orgManagement/orgManagementEdit.html]: Alex is going to send me those.

File: app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js (hunk 5 above)
Comment (adammeyer) [reply app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js]: So the parent h2 isn't clicked as well.

File: app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js (hunk 6 above)
Comment (adammeyer) [reply app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js]: What do you recommend?  I thought of having another attribute on the sub-org component, that changed and have a watch inside the component to refresh the orgs on change.  That seems hacky too.  Thoughts?

Comment (OzzieOrca) [reply app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js]: Ya... idk. You could share some refresh function with the parent component like I did in my PR with that onInit thing... In React, you would fire a Redux action and update the global state store which would rerender the components as needed.

File: app/assets/javascripts/angular/components/orgManagement/orgManagement.scss [hunk 8]
```diff
@@ -12,7 +12,7 @@ org-management {
     }
 
     .bg-pivot-gray-5 {
-        background-color: #f0f0f0;
+        background-color: $gray-5;
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/orgManagement/orgManagement.scss]: That's not what I meant. https://github.com/CruGlobal/missionhub-web/blob/master/app/assets/stylesheets/variables/colors.scss#L55 makes a class for every variable in the $pivot-colors variable. All you have to do is add a new key and it will make the class. That way all these classes stay centralized.

File: src/i18n/locales/en-US.js [hunk 9]
```diff
@@ -562,7 +562,10 @@ export default {
             show_sub: 'Show Sub-Organizations',
             none_found: 'No organizations found',
             delete: {
-                confirm: 'Are you sure you want to delete "{{org_name}}"?',
+                confirm:
+                    'Are you sure you want to delete "{{org_name}}"?' +
+                    '<br><br>' +
```
Comment (OzzieOrca) [inline src/i18n/locales/en-US.js]: This might be better as 2 different keys

File: app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js (hunk 6 above)
Comment (adammeyer) [reply app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js]: I tried a few different refresh functions and wasn't able to come up with anything.  I'm not sure what I'm doing wrong, but it would always just refresh the parent, not the sub.

Comment (OzzieOrca) [reply app/assets/javascripts/angular/components/orgManagement/orgManagementSubOrg.component.js]: I guess this is fine. If you were doing a refresh function, you'd have to write the refresh function in the child component, call an output function on the child component $onInit, save a reference to it in the parent, and then pass it to this component to call. Angular makes this too complex...
